### PR TITLE
[#1414] Add message before closing a stale PR

### LINKED ISSUE #1414: add a message when the bot closes a stale issue/PR
We can add an explanatory message when the bot closes a stale PR e.g., https://github.com/reposense/RepoSense/pull/1308#event-4220302825

### PR DESCRIPTION
**Pull request by dcshzj**
Fixes #1414.

```
The current workflow is set up such that a PR that is inactive for
30 days will be marked as stale with a comment. However, after the
7 days have elapsed with no further activity, the workflow just
closes the PR without any comment.

Let's add a comment to the PR when closing it, so that it is
clearer to the PR author on why their PR was closed.

Let's also specify only the major version for the dependency to
remain forward-compatible.
```

### CODE REVIEW

**Comment by gerhean** [comment]
LGTM but is there a reason for upgrading the version of the dependencies? Will it break anything?

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -7,20 +7,23 @@ on:
 
 jobs:
   stale:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/stale@v3.0.7
+    - uses: actions/stale@v3
```
Comment (gerhean) [inline .github/workflows/stale.yml:13]: Is there a reason for upgrading the version of the dependencies? Will it break anything? Maybe write that in your commit message too?

**Review by gerhean — COMMENTED** [review]
Otherwise LGTM

File: .github/workflows/stale.yml (hunk 1 above)
Comment (dcshzj) [reply .github/workflows/stale.yml:13]: It is to remain forward-compatible. As per semver, changes in the minor and patch versions will not cause breaking changes. I have also modified the commit message accordingly.

**Review by gerhean — COMMENTED** [review]
LGTM

**Review by gerhean — APPROVED** [review]

**Comment by gerhean** [comment]
Actually is there a way to test that this works?

**Comment by dcshzj** [comment]
> Actually is there a way to test that this works?

The only way I can think of is to create a PR in a test repository and wait for 30 + 7 days. However, I think that the code is straightforward enough to avoid over-testing.

**Comment by HCY123902** [comment]
LGTM

**Review by 0blivious — APPROVED** [review]

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -7,20 +7,23 @@ on:
 
 jobs:
   stale:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
```
Comment (Tejas2805) [inline .github/workflows/stale.yml:10]: Always better to use a specific version in my opinion. As the `latest` can keep changing.

**Review by Tejas2805 — COMMENTED** [review]
One nit. Rest LGTM.

File: .github/workflows/stale.yml (hunk 2 above)
Comment (dcshzj) [reply .github/workflows/stale.yml:10]: Yep I wanted it to change along with the use of `actions/stale@v3`, so we don't have to keep updating this.

Comment (Tejas2805) [reply .github/workflows/stale.yml:10]: It's not about having to continuously update it. If it's a fixed version, we will always know up to which version it worked well with. @fzdy1914 your opinion.

Comment (fzdy1914) [reply .github/workflows/stale.yml:10]: From my view, we should not use `an unspecified version` unless it is truly needed.

Comment (dcshzj) [reply .github/workflows/stale.yml]: I disagree in this context, but nonetheless I have reverted this change based on the consensus reached.

Comment (fzdy1914) [reply .github/workflows/stale.yml]: @dcshzj It is to avoid unwanted bugs that may happen due to the latest version while the environment changes.

**Pull request merged**